### PR TITLE
Ht 1/ht 68/logout button

### PR DESCRIPTION
**Pull request by kalibria**
Create a logout button. Call the /logout endpoint and redirect to login page. Handling errors.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1 +1 @@
-REACT_APP_BASE_URL='https://18qv4o7xm9.execute-api.eu-west-1.amazonaws.com'
+REACT_APP_BASE_URL='https://skhcma77a3.execute-api.eu-west-1.amazonaws.com'
```
Comment (ikushlianski) [inline .env]: your .env file is not in `.gitignore`.

Please include it there and remove this sensitive data from Git

File: package.json [hunk 2]
```diff
@@ -17,9 +17,11 @@
     "formik": "^2.2.9",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-hot-toast": "^2.4.0",
```
Comment (ikushlianski) [inline package.json]: are you using this lib?

File: src/auth/auth.types.ts [hunk 3]
```diff
@@ -1,5 +1,27 @@
...
+export interface MyCheckboxProps {
+  name: string;
+  [key: string]: string | FieldHookConfig<any>;
+}
+
+export interface IAlertForm {
+  message: string;
+}
+
+export interface IButtonEl {
```
Comment (ikushlianski) [inline src/auth/auth.types.ts]: IButtonEl does not seem to belong to `auth.types`. It should belong to other types, like `component.types.ts` and should not be in the auth folder. Same for other UI types in this file. This file should only contain auth-related stuff

File: src/auth/authenticationManager.ts [hunk 4]
```diff
@@ -11,10 +11,10 @@ class AuthenticationManager {
       typeof error.data === 'object' &&
       error.data !== null &&
       'status' in error.data &&
-      typeof error.data.status === 'string'
+      typeof (error.data as any).status === 'string'
```
Comment (ikushlianski) [inline src/auth/authenticationManager.ts:14]: would be great to add a comment here that this type casting happened because of TS version incompatibility with react-router. Otherwise we would not need to use this `any`.
This comment is needed in a real project to avoid unnecessary questions

File: src/auth/components/loginForm/componentsForLoginForm.tsx [hunk 5]
```diff
@@ -63,3 +55,11 @@ export const AlertForm: React.FC<IAlertForm> = ({ message }: IAlertForm) => {
...
   );
 };
+
+export const ButtonEl: React.FC<IButtonEl> = ({ text, ...props }) => {
+  return (
+    <Button variant="contained" type="submit" {...props}>
+      {text}
+    </Button>
+  );
+};
```
Comment (ikushlianski) [inline src/auth/components/loginForm/componentsForLoginForm.tsx:65]: Button element does not belong only to `LoginForm`. It would be anywhere in the app, at least that's what I think looking at the name `ButtonEl`. So please move this component to some place for generic components that are not related to a particular piece of UI

**Review by ikushlianski — CHANGES_REQUESTED** [review]

File: .env (hunk 1 above)
Comment (kalibria) [reply .env]: done

File: package.json (hunk 2 above)
Comment (kalibria) [reply package.json]: removed

File: src/auth/auth.types.ts (hunk 3 above)
Comment (kalibria) [reply src/auth/auth.types.ts]: done

File: src/auth/authenticationManager.ts (hunk 4 above)
Comment (kalibria) [reply src/auth/authenticationManager.ts:14]: done

File: src/auth/components/loginForm/componentsForLoginForm.tsx (hunk 5 above)
Comment (kalibria) [reply src/auth/components/loginForm/componentsForLoginForm.tsx:65]: done

**Review by ikushlianski — APPROVED** [review]
